collectors/python.d.plugin/edgex/edgex.chart.py

```diff
@@ -0,0 +1,312 @@
+# -*- coding: utf-8 -*-
+# Description: EdgeX Platform netdata python.d module
+# Author: Odysseas Lamtzidis (OdysLam)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Module Version: 0.2
+
+import json
+import threading
+
+from collections import namedtuple
+
+try:
+    from Queue import Queue
+except ImportError:
+    from queue import Queue
+
+from bases.FrameworkServices.UrlService import UrlService
+
+METHODS = namedtuple('METHODS', ['get_data', 'url', 'run'])
+
+ORDER = [
+    'events_throughput',
+    'readings_throughput',
+    # 'events_count',
+    'devices_number',
+    'memory_alloc',
+    'memory_malloc',
+    'memory_frees',
+    'memory_liveObjects'
+]
+
+# default module values (can be overridden per job in `config`)
+update_every = 5
+
+CHARTS = {
+    'events_throughput':{
+        'options': [None, 'Events and Readings (readings are a certain type of event) per second in the EdgeX platform', 'events/s',
+                    'events throughput','edgex.events', 'line'],
+        'lines': [
+            ['events', None, 'incremental'],
+            ['readings', None, 'incremental']
+        ]
+    },
+    # 'readings_throughput':{
+    #     'options': [None, 'Readings per second in the EdgeX platform', 'readings/s', 'throughput', 'edgex.readings',
+    #                 'line'],
+    #     'lines': [
+    #         ['readings', None, 'incremental']
+    #     ]
+    # },
+    'events_count':{
+        'options': [None, 'Total number of events and readings in the EdgeX platform', 'events', 'events count', 'edgex.events_readings_abs',
+                    'line'],
+        'lines': [
+            ['readings', None, 'absolute'],
+            ['events', None, 'absolute']
+        ]
+    },
+    'devices_number':{
+        'options': [None, 'Number of registered devices in the EdgeX platform', 'devices', 'devices count', 'edgex.devices_number',
+                    'line'],
+        'lines': [
+            ['registered_devices', None, 'absolute'],
+        ]
+    },
+    'memory_alloc':{
+        'options': [None, 'Alloc: currently allocated number of bytes on the heap of each EdgeX service. Provided by the Golang runtime Package.',
+                    'bytes', 'memory metrics', 'edgex.memory_alloc', 'line'],
+        'lines': [
+            ['core_data_alloc', None, 'absolute'],
+            ['core_metadata_alloc', None, 'absolute'],
+            ['core_command_alloc', None, 'absolute'],
+            ['support_logging_alloc', None, 'absolute']
```

Comment:
i think we need names here

> ['core_data_alloc', None, 'absolute'] =>  ['core_data_alloc', 'core_data', 'absolute']

Reply:
To be honest, I had a hard time understanding the proper setup of the charts. I will make the change as I really don't have an opinion on how it should be.

Reply:
Check docs, it should make things clear

https://docs.netdata.cloud/collectors/plugins.d/#chart

Reply:
Please believe me when I say that using the docs, the charts configuration was the most ambiguous part of the plugin development.